### PR TITLE
Add blobs search from gitlab search Api

### PR DESCRIPTION
**Pull request by Toa741**
- https://docs.gitlab.com/ee/api/search.html#advanced-search-api
- Update GitLabDockerVersion in tests as the comment suggested
- Unable to add tests (scoped search requires elasticsearch to be installed separately)
- Not sure mocks can be easily implemented either

### CODE REVIEW

File: NGitLab/Impl/SearchClient.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using NGitLab.Models;
+
+namespace NGitLab.Impl
+{
+    public class SearchClient : ISearchClient
```
Comment (meziantou) [inline NGitLab/Impl/SearchClient.cs]:
```suggestion
    internal sealed class SearchClient : ISearchClient
```

File: NGitLab/ISearchClient.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿using NGitLab.Models;
+
+namespace NGitLab
+{
+    public interface ISearchClient
+    {
+        GitLabCollectionResponse<SearchBlob> GetBlobs(SearchQuery query);
```
Comment (meziantou) [inline NGitLab/ISearchClient.cs]: I think other `GitLabCollectionResponse` methods use the `Async` suffix.
```suggestion
        GitLabCollectionResponse<SearchBlob> GetBlobsAsync(SearchQuery query);
```

File: NGitLab/Impl/SearchClient.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            url = Utils.AddParameter(url, "search", query.Search);
+
+            if (query.PerPage.HasValue)
+            {
+                url = Utils.AddParameter(url, "per_page", query.PerPage);
+            }
+
+            if (query.Confidential.HasValue)
+            {
+                if (!string.Equals(scope, "issues", StringComparison.Ordinal))
```
Comment (meziantou) [inline NGitLab/Impl/SearchClient.cs]: Can't we let GitLab do the check?

File: NGitLab/Impl/SearchClient.cs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                {
+                    throw new NotSupportedException("Only allowed value for OrderBy is 'created_at'");
+                }
+
+                url = Utils.AddOrderBy(url, query.OrderBy);
+            }
+
+            if (!string.IsNullOrEmpty(query.Sort))
+            {
+                if (!string.Equals(query.Sort, "asc", StringComparison.Ordinal) &&
```
Comment (meziantou) [inline NGitLab/Impl/SearchClient.cs]: I don't think we need a validation here. Let's GitLab do the job.

File: NGitLab/SearchQuery.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
+﻿namespace NGitLab
+{
+    public class SearchQuery
```
Comment (meziantou) [inline NGitLab/SearchQuery.cs]:
```suggestion
    public sealed class SearchQuery
```

File: NGitLab/Impl/SearchClient.cs (hunk 3 above)
Comment (Toa741) [reply NGitLab/Impl/SearchClient.cs]: Yes, it will just ignore if not of the right type

File: NGitLab/PublicAPI.Unshipped.txt [hunk 6]
```diff
@@ -669,8 +675,8 @@ NGitLab.Impl.RunnerClient.Register(NGitLab.Models.RunnerRegister request) -> NGi
 NGitLab.Impl.RunnerClient.RunnerClient(NGitLab.Impl.API api) -> void
 NGitLab.Impl.RunnerClient.this[int id].get -> NGitLab.Models.Runner
 NGitLab.Impl.RunnerClient.Update(int runnerId, NGitLab.Models.RunnerUpdate runnerUpdate) -> NGitLab.Models.Runner
+NGitLab.Impl.SearchClient.GetBlobs(NGitLab.SearchQuery query) -> NGitLab.GitLabCollectionResponse<NGitLab.Models.SearchBlob>
```
Comment (meziantou) [inline NGitLab/PublicAPI.Unshipped.txt]:
```suggestion
NGitLab.Impl.SearchClient.GetBlobsAsync(NGitLab.SearchQuery query) -> NGitLab.GitLabCollectionResponse<NGitLab.Models.SearchBlob>
```

**Pull request merged**